scripts/build-packages.js

```diff
@@ -0,0 +1,27 @@
+const path = require("path");
+const fs = require("fs");
+const shell = require("shelljs");
+
+shell.cd(path.join(__dirname, ".."));
+
+const packages = fs
+    .readdirSync(path.join(__dirname, "..", "packages"))
+    .filter(name => !name.startsWith("."));
+
+for (const package of packages) {
+    const pkgPath = path.join(__dirname, "..", "packages", package);
+    const pkgJSONPath = path.join(pkgPath, "package.json");
+
+    if (fs.existsSync(pkgJSONPath)) {
+        const pkgJSON = require(pkgJSONPath);
+
+        // some packages may not have build command
+        if (pkgJSON && pkgJSON.scripts && pkgJSON.scripts.build) {
+            console.log(`\nbuilding "/packages/${package}"\n`);
+
+            if (shell.exec(`yarn --cwd "${pkgPath}" build`).code !== 0) {
+                shell.exit(1);
+            }
```

Comment:
The original exit code should not be changed.

```suggestion
            const buildExitCode = shell.exec(`yarn --cwd "${pkgPath}" build`).code;
            if (buildExitCode !== 0) {
                shell.exit(buildExitCode);
            }
```

Reply:
This code style is from [shelljs example](https://github.com/shelljs/shelljs#examples).

Reply:
This is just **example**, We should modify it according to our own situation.

Reply:
Do we have a code style rule for this one? If not I think they offer similar readability while the example code is cleaner.

Intermediate variable works great for long or complex expressions. The downside is that to read the assignment result for variable name it forces one to move reading direction from right to left which is against the code natural direction.

So for short and clean expression, keeping the natural reading direction is easier to read in my opinion.

Reply:
Then we can refer to the design of golang, where we can change the variable name to a single word, such as: `c`

In my opinion, my focus is "should not swallow the original exit code", so the problem of code style, as long as you feel no problem, is OK

Reply:
Oh I see what you mean. Yeah the exit code should be kept.

Reply:
Done.